acceptance/lib/pxp-agent/test_helper.rb

```diff
@@ -132,16 +194,43 @@ def rpc_blocking_request(broker, targets,
     end
   rescue Timeout::Error
     mutex.synchronize do
-      if !have_all_rpc_responses?
-        raise "Didn't receive all PCP responses when requesting puppet run on #{targets}. Responses received were: #{responses.to_s}"
+      if !have_all_rpc_responses?(targets, responses)
+        raise "Didn't receive all PCP responses when requesting #{pxp_module} #{action} on #{targets}. Responses received were: #{responses.to_s}"
       end
     end
   end # wait for message
 
-  em.exit
+  em.stop unless em.stop?
```

Comment:
You should be stopping eventmachine here, not the thread it's runloop is in.  `EM.stop`